packages/core/core-flows/src/order/workflows/cancel-order-fulfillment.ts

```diff
@@ -114,44 +125,120 @@ function prepareCancelOrderFulfillmentData({
   order: OrderDTO
   fulfillment: FulfillmentDTO
 }) {
+  const lineItemIds = new Array(
+    ...new Set(fulfillment.items.map((i) => i.line_item_id))
+  )
+
   return {
     order_id: order.id,
     reference: Modules.FULFILLMENT,
     reference_id: fulfillment.id,
-    items: fulfillment.items!.map((i) => {
+    items: lineItemIds.map((lineItemId) => {
+      // find order item
+      const orderItem = order.items!.find((i) => i.id === lineItemId)
+      // find inventory items
+      const iitems = (orderItem as any)!.variant?.inventory_items
```

Comment:
nit: I've seen `(orderItem as any).variant` a few times in this PR, would it make sense to update the type to reference it?